scanner/cmd/scannerctl/main.go

```diff
@@ -21,6 +25,9 @@ import (
 func main() {
 	certsPath := flag.String("certs", "", "Path to directory containing scanner certificates.")
 	basicAuth := flag.String("auth", "", "Use basic auth to authenticate with registries.")
+	imageDigest := flag.String("digest", "", "Use the specified image digest in the image "+
```

Comment:
Please refer to [ADR-0001](https://github.com/stackrox/stackrox/blob/master/scanner/decisions/0001-scannerv4-apis.md):

> ScannerV4 will use "index reports" and "vulnerability reports" data models, similar to ClairCore's. Both types will link to a "scannable resource" (e.g., a container image) using an ID created by Scanner clients (i.e. hash_id). This is a string that uniquely identifies the resource's manifest.

So, the "hash_id" is a unique identifier for the scanned resource. It follows:

> To avoid conflicts between different resources and allow changes to how the IDs are generated, Central and Sensor (the existing clients) will namespace them with `/v1/<resource>/`. Initially, `/v1/containerimage/<image-digest>` will be supported for container images.

So, the `hash_id` will be prefixed with the image digest for container images.

Now, the flag in `scannerctl` allows the user to select the image digest used to populate `<image-digest>` in `/v1/containerimage/<image-digest>`. This will not be necessary for normal Scanner operations. Central and Sensor will almost certainly use the image digest retrieved from the image metadata call they perform before requesting the image to be scanned. Nonetheless, `scannerctl` allows the caller to skip calling the registry and specifying the digest directly.

Others already suggested using the `@<image-digest>` in the command line param if present, which is an excellent addition, and I captured it in a different story.

I hope this clarifies.

Reply:
I see now, thanks